src/database/DataTypes/ItemBatch.js

```diff
@@ -145,6 +150,20 @@ export class ItemBatch extends Realm.Object {
 
     return newLocationId !== currentLocationId;
   }
+
+  applyLocation(database, newLocation) {
+    if (!this.shouldApplyLocation(newLocation)) return null;
+
+    this.location = newLocation;
+
+    return createRecord(database, 'LocationMovement', this, newLocation);
```

Comment:
This is called after a stocktake or transaction are finalised - locking in the location movement. applying a location: Adds a `LocationMovement` record so there is a ledger of location movements and at what time. Also sets the location field to the current location (is already on item batch so just left it .. )